kore/src/Kore/ASTVerifier/SentenceVerifier.hs

```diff
@@ -260,6 +266,31 @@ verifySymbolSentence indexedModule sentence =
     findSort = findIndexedSort indexedModule
     sortParams = (symbolParams . sentenceSymbolSymbol) sentence
 
+    verifyConstructorNotInHookedSort :: Either (Error VerifyError) ()
+    verifyConstructorNotInHookedSort =
+        let
+            symbol = symbolConstructor $ sentenceSymbolSymbol sentence
+            attributes = sentenceSymbolAttributes sentence
+            resultSort = sentenceSymbolResultSort sentence
+            resultSortId = getSortId resultSort
+
+            ctor =
+                Attribute.constructorAttribute  `elem` getAttributes attributes
+            sort =
+                join
```

Comment:
Would this be clearer using `do`-notation instead of `join`? I had to read this a few times to figure out what is being `join`ed.